src/components/Button/Button.tsx

```diff
@@ -1,28 +1,44 @@
 import PropTypes from 'prop-types'
-import React from 'react'
+import React, { ReactNode, CSSProperties } from 'react'
 
-import { UIComponent, childrenExist, customPropTypes } from '../../lib'
+import { UIComponent, childrenExist, customPropTypes, IRenderResultConfig } from '../../lib'
 import buttonRules from './buttonRules'
 import buttonVariables from './buttonVariables'
 
+export type ButtonType = 'primary' | 'secondary'
+
+export interface IButtonProps {
+  as?: string
+  children?: ReactNode
+  circular?: boolean
+  className?: string
+  content?: ReactNode
+  fluid?: boolean
+  style?: CSSProperties
+  type?: ButtonType
+}
+
 /**
  * A button.
  * @accessibility This is example usage of the accessibility tag.
  * This should be replaced with the actual description after the PR is merged
  */
-class Button extends UIComponent<any, any> {
-  static displayName = 'Button'
+class Button extends UIComponent<IButtonProps, any> {
+  public static displayName = 'Button'
```

Comment:
TS uses [public by default](https://www.typescriptlang.org/docs/handbook/classes.html#public-by-default) in classes.  Let's keep the code concise.

Reply:
I'm aware, but I don't like that feature; I feel that with `public by default` developers are prone to exposing methods that should be private just because the habit of adding a modifier is not enforced; I don't mind reverting tho'

Reply:
Sounds good.  Want to make sure it is addressed properly as well.  If you feel very strongly for this topic, please raise an RFC and let's have some good conversation around it.